src/Paprika.Tests/Merkle/NodeTests.cs

```diff
@@ -0,0 +1,234 @@
+using System.Reflection.Emit;
+using NUnit.Framework;
+using Paprika.Crypto;
+using Paprika.Data;
+using Paprika.Merkle;
+
+namespace Paprika.Tests.Merkle;
+
+public class NodeTests
+{
+    [Test]
+    [TestCase(typeof(Node.Header), 1)]
+    [TestCase(typeof(Node.Branch), 64)]
+    [TestCase(typeof(Node.Leaf), 64)]
+    [TestCase(typeof(Node.Extension), 32)]
+    public void Struct_size(Type type, int expectedSize)
+    {
+        Assert.That(GetSizeOfType(type), Is.EqualTo(expectedSize));
+    }
+
+    [Test]
+    [TestCase(Node.Type.Branch, true)]
+    [TestCase(Node.Type.Branch, false)]
+    [TestCase(Node.Type.Leaf, true)]
+    [TestCase(Node.Type.Leaf, false)]
+    [TestCase(Node.Type.Extension, true)]
+    [TestCase(Node.Type.Extension, false)]
+    public void Header_properties(Node.Type nodeType, bool isDirty)
+    {
+        var header = new Node.Header(nodeType, isDirty);
+
+        Assert.That(header.NodeType, Is.EqualTo(nodeType));
+        Assert.That(header.IsDirty, Is.EqualTo(isDirty));
+    }
+
+    [Test]
+    [TestCase(Node.Type.Leaf, false)]
+    [TestCase(Node.Type.Extension, false)]
+    [TestCase(Node.Type.Branch, false)]
+    [TestCase(Node.Type.Leaf, true)]
+    [TestCase(Node.Type.Extension, true)]
+    [TestCase(Node.Type.Branch, true)]
+    public void Node_header_read_write(Node.Type nodeType, bool isDirty)
+    {
+        var expected = new Node.Header(nodeType, isDirty);
+        Span<byte> buffer = stackalloc byte[Node.Header.Size];
+
+        _ = expected.WriteTo(buffer);
+        _ = Node.Header.ReadFrom(buffer, out var actual);
+
+        Assert.That(actual.Equals(expected), $"Expected {expected.ToString()}, got {actual.ToString()}");
+    }
+
+    [Test]
+    public void Branch_properties()
+    {
+        ushort nibbles = 0b0000_0000_0000_0000;
```

Comment:
It actually was helpful since I initially had explicit struct layouts everywhere, and these constructor tests were used to check if the files were being loaded at all.

We can drop it if you think it's not useful, though.